appcues/src/main/java/com/appcues/Appcues.kt

```diff
@@ -277,21 +272,6 @@ class Appcues internal constructor(koinScope: Scope) {
     // observe and re-broadcast tracking data as desired.
     @VisibleForTesting
     internal fun publishTracking(data: TrackingData) {
-
-        fun EventRequest.screenTitle(): String? =
-            attributes[ActivityRequestBuilder.SCREEN_TITLE_ATTRIBUTE] as? String
-
-        analyticsListener?.let { listener ->
-            when (data.type) {
-                EVENT -> data.request.events?.forEach {
-                    listener.trackedAnalytic(EVENT, it.name, it.attributes, data.isInternal)
-                }
-                IDENTIFY -> listener.trackedAnalytic(IDENTIFY, storage.userId, data.request.profileUpdate, data.isInternal)
-                GROUP -> listener.trackedAnalytic(GROUP, storage.groupId, data.request.groupUpdate, data.isInternal)
-                SCREEN -> data.request.events?.forEach {
-                    listener.trackedAnalytic(SCREEN, it.screenTitle(), it.attributes, data.isInternal)
-                }
-            }
-        }
+        analyticsPublisher.publish(analyticsListener, data)
```

Comment:
could move this call up to line 90 and eliminate the need for this function entirely?

Reply:
it might limit our testing capability if we do so

Reply:
I don't think it would limit it really. I think we'd just need to update the test to call `analyticsPublisher.publish` instead of `appcues.publishTracking`, and it would effectively test the same logic for the formatting of the analytics listener output.